Source/Core/DolphinQt/Config/Mapping/IOWindow.cpp

```diff
@@ -448,8 +476,10 @@ void IOWindow::OnDialogButtonPressed(QAbstractButton* button)
     return;
   }
 
-  m_reference->SetExpression(m_expression_text->toPlainText().toStdString());
-  m_controller->UpdateSingleControlReference(g_controller_interface, m_reference);
+  const auto lock = m_controller->GetStateLock();
+
+  UpdateExpression(m_expression_text->toPlainText().toStdString());
+  m_original_expression = m_reference->GetExpression();
```

Comment:
With this new `m_original_expression` state maybe we can just kill the "Apply" button entirely now that it applies in real time?

Reply:
I added a commit that removes the apply button, but I don't really care whether it stays or not. If we decide to keep it we can just drop the commit.